espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/handlers/RequestPermissionsHandler.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appium.espressoserver.lib.handlers;
+
+import androidx.test.runner.permission.PermissionRequester;
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException;
+import io.appium.espressoserver.lib.model.RequestPermissionsParams;
+
+public class RequestPermissionsHandler implements RequestHandler<RequestPermissionsParams, Void> {
+    @Override
+    public Void handle(RequestPermissionsParams params) throws AppiumException {
+        final PermissionRequester permissionRequester = new PermissionRequester();
+        for (final String permission: params.getPermissions()) {
+            try {
+                permissionRequester.addPermissions(permission);
```

Comment:
What about defending  `params.getPermissions()` as `String[]`?
Then, we can get rid of `for` and can call ` permissionRequester.addPermissions(params.getPermissions());`, simply.

Reply:
`addPermissions` accepts Strings... I don't think it accepts `String[]`

Reply:
ah, I have simply implemented the same code as:

```kotlin
    private fun grantPermissions(permissions: Array<String>) {
        val requester = PermissionRequester()
        requester.addPermissions(*permissions)
        requester.requestPermissions()
    }
```

The converted to Java code was:

```java
   private final void grantPermissions(String[] permissions) {
      PermissionRequester requester = new PermissionRequester();
      requester.addPermissions((String[])Arrays.copyOf(permissions, permissions.length));
      requester.requestPermissions();
   }
```

So I thought so.